paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/Form.kt

```diff
@@ -0,0 +1,149 @@
+import com.stripe.android.paymentsheet.elements.common.Element
+import com.stripe.android.paymentsheet.elements.common.Section
+import com.stripe.android.paymentsheet.elements.common.TextField
+import com.stripe.android.paymentsheet.elements.common.TextFieldElement
+import com.stripe.android.paymentsheet.elements.country.CountryConfig
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.map
+
+// TODO: Manadate type
+// TODO: Save for future usage.
+
+@ExperimentalAnimationApi
+@Composable
+internal fun Form(
+    formViewModel: FormViewModel,
+) {
+    val form = formViewModel.formDataObject
+
+    // There is only a single field in a section, some of those might not require focus like
+    // country so we will create an extra one
+    var focusRequesterIndex = 0
+    val focusRequesters = List(formViewModel.getNumberTextFieldElements()) { FocusRequester() }
+
+    Column(
+        modifier = Modifier
+            .fillMaxWidth(1f)
+            .padding(16.dp)
+    ) {
+        form.sections.forEach { section ->
+            val element = formViewModel.getElement(section.field)
+            val error by element.errorMessage.asLiveData().observeAsState(null)
+            val sectionErrorString =
+                error?.let { stringResource(it, stringResource(element.label)) }
+
+            Section(sectionErrorString) {
+                if (element is TextFieldElement) {
+                    TextField(
+                        textFieldElement = element,
+                        myFocus = focusRequesters[focusRequesterIndex],
+                        nextFocus = if (focusRequesterIndex == focusRequesters.size - 1) {
+                            null
+                        } else {
+                            focusRequesters[focusRequesterIndex + 1]
+                        },
+                    )
+                    focusRequesterIndex++
+                } else if (element is DropdownElement) {
+                    DropDown(element)
+                }
+            }
+        }
+    }
+}
+
+class FormViewModel(
+    val formDataObject: FormDataObject
+) : ViewModel() {
+    class Factory(
+        val formDataObject: FormDataObject
+    ) : ViewModelProvider.Factory {
+        @Suppress("UNCHECKED_CAST")
+        override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+            return FormViewModel(formDataObject) as T
+        }
+    }
+
+    private val paramKey: MutableMap<String, Any?> = formDataObject.paramKey
+    private val types: List<Field> = formDataObject.allTypes
+    private val elementMap = mutableMapOf<Field, Element>()
+    fun getNumberTextFieldElements() = elementMap.filter { it.value is TextFieldElement }.size
+
+    internal fun getElement(type: Field) = requireNotNull(elementMap[type])
+
+    private val isComplete: Flow<Boolean>
+    private val params: Flow<MutableMap<String, Any?>>
+    val paramMapFlow: Flow<MutableMap<String, Any?>?>
```

Comment:
I think a better pattern would to be to have `FormViewModel` emit a `data class FormData`, and have `FormData` take care of converting typed data into a `Map`.